pkg/controller/upgradeconfig/cluster_upgrader.go

```diff
@@ -250,40 +251,61 @@ func CommenceUpgrade(c client.Client, upgradeConfig *upgradev1alpha1.UpgradeConf
 	return true, nil
 }
 
-// TODO This create the maint window for control plane
-func CreateControlPlaneMaintWindow(c client.Client, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
+// Create the maintenance window for control plane
+func CreateControlPlaneMaintWindow(c client.Client, m maintenance.Maintenance, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
+	err := m.Start(time.Now().Add(time.Minute * 90))
```

Comment:
I'll raise it now because it's in my mind, but I wonder if there's an opportunity for us to have a better place for storing the static 'control plane' and 'worker' time periods - hidden in code makes it easy to miss - arguably the same reason why the maintenance window logic for the previous upgrade script was not easily discernable.

Not sure about the best way. A configmap for the operator maybe? Or an amendment to the UpgradeConfig CRD. But something I personally think would be a good opportunity to address before we hit Milestone 1. What do you think?

Reply:
@mrbarge  Definite +1 to extracting config.
I would go with a configmap but was wondering about how its created/managed. Would this also be an object in the `SelectorSyncSet`?

Reply:
@jroche yes that's what I'd be thinking. Like, maybe a ConfigMap which is managed as part of the `managed-cluster-config` setup for the operator, which would then get incorporated into a `SelectorSyncSet` when the next release of `managed-cluster-config` is built.